src/types/zoneddatetime.jl

```diff
@@ -118,7 +118,8 @@ end
 
 # Parsing constructor. Note we typically don't support passing in time zone information as a
 # string since we cannot do not know if we need to support resolving ambiguity.
-function ZonedDateTime(y::Int64, m::Int64, d::Int64, h::Int64, mi::Int64, s::Int64, ms::Int64, tz::AbstractString)
+# Month Union issue: https://github.com/JuliaTime/TimeZones.jl/issues/187#issuecomment-473012078
+function ZonedDateTime(y::Int64, m::Union{Int32, Int64}, d::Int64, h::Int64, mi::Int64, s::Int64, ms::Int64, tz::AbstractString)
```

Comment:
Can we add a test for this? AppVeyor should run on 32-bit.

Reply:
Wouldn't it be better to just declare the arguments as `Integer`?   The `DateTime` constructor will convert them as needed, and the compiler will type-specialize the values anyway.

Reply:
I'd honestly prefer not to have this constructor (see the note in the code). Keeping the types very tight here discourages general usage. The only reason this constructor exists is for interoperability with the Dates parsing code.